cmd/metal-api/internal/datastore/event.go

```diff
@@ -35,3 +38,41 @@ func (rs *RethinkStore) CreateProvisioningEventContainer(ec *metal.ProvisioningE
 func (rs *RethinkStore) UpsertProvisioningEventContainer(ec *metal.ProvisioningEventContainer) error {
 	return rs.upsertEntity(rs.eventTable(), ec)
 }
+func (rs *RethinkStore) ProvisioningEventForMachine(machineID, event, message string) (*metal.ProvisioningEventContainer, error) {
+	ec, err := rs.FindProvisioningEventContainer(machineID)
+	if err != nil && !metal.IsNotFound(err) {
+		return nil, err
+	}
+
+	if ec == nil {
+		ec = &metal.ProvisioningEventContainer{
+			Base: metal.Base{
+				ID: machineID,
+			},
+			Liveliness: metal.MachineLivelinessAlive,
+		}
+	}
+	now := time.Now()
+	ec.LastEventTime = &now
+
+	ev := metal.ProvisioningEvent{
+		Time:    now,
+		Event:   metal.ProvisioningEventType(event),
+		Message: message,
+	}
+	if ev.Event == metal.ProvisioningEventAlive {
+		zapup.MustRootLogger().Sugar().Debugw("received provisioning alive event", "id", ec.ID)
```

Comment:
When we already touch this, can we pass a logger, too? Would be nice to reduce the usage of zapup.

Reply:
done